containers/Containerfile.alpine

```diff
@@ -0,0 +1,5 @@
+FROM docker.io/library/alpine:latest
+
+# tmt requires `bash` to be installed
+# tests expect `tar` and `flock` to be installed
+RUN apk add --no-cache bash flock tar
```

Comment:
I don't understand why a custom image with all these preinstalled is needed. Yes, `/usr/bin/flock` is needed to run any tmt tests in the environment, but, the point of package manager and essential requirements is to install `/usr/bin/flock` before running a test. Preinstalling it makes it very comfy for `apk` package manager, it doesn't have to move a finger :)

Plus, and this is what's puzzling, for *unit tests* of package managers, `/usr/bin/flock` does not have to be installed, those are not tmt tests. The same applies to `bash` and `tar` too, package manager unit tests check whether basic actions like `install` or `uninstall` work, and run no tmt tests at all.

@thrix what fails if you try unit tests with the bare alpine image?

Reply:
I guess the image is needed because we rely on `bash` for all the shell script operations? The other two packages though could/should be removed, I'd say.

Reply:
I don't know about `bash` either - what those package manager unit tests do is that they spawn a container, whatever its entry point might be, and run `dnf install -y foo` in this container. Yes, `bash` is something tmt enforces for shell scripts, but these tests are nowhere near it...

Reply:
@happz all those deps were added to make the tests behave the same, I can definitely shuffle with `tar`, play with `flock`, but `bash` is what we run in container. If bash is not present it fails. Let me gather some logs.

Reply:
```
DEBUG    tmt:log.py:723 Run command: podman exec 8495dc6792e1c1a43b35ca47cfb4424451e15b51eeeeca000c396b86eb881c23 /bin/bash -c 'cat /etc/os-release'
DEBUG    tmt:log.py:723 environment
DEBUG    tmt:log.py:723 Command event: 1.595e-06 waiting for process to finish
DEBUG    tmt:log.py:723 err: Error: crun: executable file `/bin/bash` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found
DEBUG    tmt:log.py:723 Command event: 0.05496 waiting for process completed
DEBUG    tmt:log.py:723 Command event: 0.05505 waiting for stream readers
DEBUG    tmt:log.py:723 Command event: 0.05508 stdout reader done
DEBUG    tmt:log.py:723 Command event: 0.0551 stderr reader done
DEBUG    tmt:log.py:723 Command returned '127'.
DEBUG    tmt:log.py:723 Run command: podman exec 8495dc6792e1c1a43b35ca47cfb4424451e15b51eeeeca000c396b86eb881c23 /bin/bash -c 'cat /etc/lsb-release'
```

Reply:
@happz if we want to address it, we would have to use `sh` in case of `alpine`. I agree it would be more clean, but the question is what we loose.

Reply:
I see, now I get it. I forgot about `/bin/bash` being added to every command.

How about keeping just the `bash`, skipping the rest? `flock` should be pulled in as an essential requirement, and if `tar` is needed but not installed, we miss the essential requirement somewhere.

Plus, I'd propose different tag, would `-t alpine` collide with regular, not-tailored-for-tmt alpine images? How about something like `alpine:tmt-unit-tests`?

Reply:
@happz what is the point to make the tests differ if we already need to add `bash` ?

Reply:
I will check on `flock`, but I believe if I did not add it, it was failing :) so not sure about that pulled in automatically part ;)

Reply:
Two different things: it's pulled in for *tmt* tests, not for *unit* tests. But unit tests use `/usr/bin/flock` only in the test that checks presence script generation - I guess that's why they failed, because the filepath was indeed missing. But you may easily pick any other file, e.g. `/bin/ls`, and update the test. `flock` is preinstalled in other distros, so, it works out of the box, but I wouldn't see using the same path in this test as mandatory. Check the rawhide, it uses `/usr/bin/arch`, because `/usr/bin/flock` leads to unstable paths and RPMs get renamed in rawhide. It's really just a path we know exists (or not) and we want to check the package manager find the right package.

Reply:
I can definitely fix the tag ....